src/components/InfoBanner/InfoBanner.tsx

```diff
@@ -0,0 +1,165 @@
+    description: string;
+    variant?: BannerVariants;
+    emphasized?: boolean;
+    linkText?: string;
+    linkUrl?: string;
+}
+
+export type BannerVariants = 'info' | 'success' | 'warning' | 'error';
+
+interface BoxWithVariant extends BoxProps {
+    variant: BannerVariants;
+    emphasized: boolean;
+}
+
+const bannerVariants = styledVariant({
+    variants: {
+        info: {
+            background: get('semanticColors.background.info'),
+            borderColor: get('semanticColors.border.info')
+        },
+        success: {
+            background: get('semanticColors.background.success'),
+            borderColor: get('semanticColors.border.success')
+        },
+        warning: {
+            background: get('semanticColors.background.warning'),
+            borderColor: get('semanticColors.border.warning')
+        },
+        error: {
+            background: get('semanticColors.background.danger'),
+            borderColor: get('semanticColors.border.danger')
+        }
+    }
+});
+
+const emphasizedBannerVariants = styledVariant({
+    variants: {
+        info: {
+            background: get('semanticColors.background.infoEmphasized'),
+            borderColor: get('semanticColors.border.infoEmphasized')
+        },
+        success: {
+            background: get('semanticColors.background.successEmphasized'),
+            borderColor: get('semanticColors.border.successEmphasized')
+        },
+        warning: {
+            background: get('semanticColors.background.warningEmphasized'),
+            borderColor: get('semanticColors.border.warningEmphasized')
+        },
+        error: {
+            background: get('semanticColors.background.dangerEmphasized'),
+            borderColor: get('semanticColors.border.dangerEmphasized')
+        }
+    }
+});
+
+const iconColorVariants = styledVariant({
+    variants: {
+        info: {
+            color: get('semanticColors.icon.action')
+        },
+        success: {
+            color: get('semanticColors.icon.success')
+        },
+        warning: {
+            color: get('semanticColors.icon.primary')
+        },
+        error: {
+            color: get('semanticColors.icon.danger')
+        }
+    }
+});
+
+const emphasizedIconColorVariants = styledVariant({
+    variants: {
+        info: {
+            color: get('semanticColors.icon.primaryInverted')
+        },
+        success: {
+            color: get('semanticColors.icon.primaryInverted')
+        },
+        warning: {
+            color: get('semanticColors.icon.primary')
+        },
+        error: {
+            color: get('semanticColors.icon.primaryInverted')
+        }
+    }
+});
+
+export const MinWidthAlertIcon = styled(AlertIcon)`
+    min-width: 1rem;
+`;
+
+const RoundedBox = styled(Box)<BoxWithVariant>`
+    display: flex;
+    flex-direction: row;
+    justify-content: flex-start;
+    border-radius: 0.5rem;
+    border: 0.0625rem solid;
+    padding: 0.5rem 1rem 0.5rem 0.5rem;
```

Comment:
Do you think we can use **Spaces** from theme instead of hard coded values here?